Ntlm.cs

```diff
@@ -630,7 +663,11 @@ private unsafe byte[] ProcessChallengeMessage(byte[] blob)
 
             // TBD calculate flags.
             response[0].Header.MessageType = MessageType.Authenticate;
-            response[0].Flags = Flags.NegotiateNtlm | Flags.NegotiateNtlm2 | Flags.NegotiateUnicode | Flags.TargetName | Flags.TargetTypeServer | Flags.NegotiateTargetInfo;
+            response[0].Flags =
+                Flags.NegotiateNtlm2 | Flags.NegotiateNtlm | Flags.NegotiateUnicode | Flags.TargetName |
+                Flags.NegotiateVersion | Flags.NegotiateKeyExchange | Flags.Negotiate128 | Flags.NegotiateTargetInfo |
+                Flags.NegotiateAlwaysSign | Flags.NegotiateSign;
+            response[0].Version = new Version { VersionMajor = 6, VersionMinor = 1, ProductBuild = 7600, CurrentRevision = 15 };
```

Comment:
Should we  take this in construction as optional? I make it static property?

Reply:
Static property is a good idea.